precovery/orbit.py

```diff
@@ -0,0 +1,443 @@
+        api_password = getenv("PRECOVERY_API_SINGLEORBIT_PASSWORD")
+
+        if not (precovery_singleorbit_url and api_username and api_password):
+            raise ValueError(
+                """one of required environment variables unset, expecting PRECOVERY_API_SINGLEORBIT_URL,
+            PRECOVERY_API_SINGLEORBIT_USERNAME, PRECOVERY_API_SINGLEORBIT_PASSWORD"""
+            )
+
+        if self._orbit_type == OrbitElementType.KEPLERIAN:
+            orbit_type = "kep"
+            state_vector = {
+                "a": self._state_vector[0][1],
+                "e": self._state_vector[0][2],
+                "i": self._state_vector[0][3],
+                "an": self._state_vector[0][4],
+                "ap": self._state_vector[0][5],
+                "ma": self._state_vector[0][6],
+            }
+        elif self._orbit_type == OrbitElementType.COMETARY:
+            orbit_type = "com"
+            state_vector = {
+                "q": self._state_vector[0][1],
+                "e": self._state_vector[0][2],
+                "i": self._state_vector[0][3],
+                "an": self._state_vector[0][4],
+                "ap": self._state_vector[0][5],
+                "tp": self._state_vector[0][6],
+            }
+        elif self._orbit_type == OrbitElementType.CARTESIAN:
+            orbit_type = "car"
+            state_vector = {
+                "x": self._state_vector[0][1],
+                "y": self._state_vector[0][2],
+                "z": self._state_vector[0][3],
+                "vx": self._state_vector[0][4],
+                "vy": self._state_vector[0][5],
+                "vz": self._state_vector[0][6],
+            }
+        else:
+            raise ValueError("orbit type improperly defined %r" % self._orbit_type)
+
+        # Compile request dictionary
+
+        if self._epoch_timescale == EpochTimescale.UTC:
+            scale = "utc"
+        elif self._epoch_timescale == EpochTimescale.UT1:
+            scale = "ut1"
+        elif self._epoch_timescale == EpochTimescale.TT:
+            scale = "tt"
+        elif self._epoch_timescale == EpochTimescale.TAI:
+            scale = "tai"
+        mjd_tdb = Time(self._epoch, scale=scale, format="mjd").tdb.mjd
+        post_json = {
+            "tolerance": tolerance,
+            "orbit_type": orbit_type,
+            "mjd_tdb": mjd_tdb,
+        }
+        post_json = post_json | state_vector
+        if max_matches:
+            post_json["max_matches"] = max_matches
+        if start_mjd:
+            post_json["start_mjd"] = start_mjd
+        if end_mjd:
+            post_json["end_mjd"] = end_mjd
+        if window_size:
+            post_json["window_size"] = window_size
+
+        precovery_req = req.post(
+            precovery_singleorbit_url, json=post_json, auth=(api_username, api_password)
+        )
+        return precovery_req.json()
+
+
+class Ephemeris:
```

Comment:
Would love an explanation of what the ephemeris represents. I have a hunch but want to be accurate in my mental model.